tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -126,35 +129,64 @@ def test_non_ocs_taint_and_tolerations(self):
         )
         ocsini_obj.patch(params=param, format_type="merge")
 
-        # Add tolerations to the configmap rook-ceph-operator-config
-        configmap_obj = ocp.OCP(
-            kind=constants.CONFIGMAP,
-            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
-            resource_name=constants.ROOK_OPERATOR_CONFIGMAP,
-        )
-        toleration = configmap_obj.get().get("data").get("CSI_PLUGIN_TOLERATIONS")
-        toleration += (
-            '\n- key: xyz\n  operator: Equal\n  value: "true"\n  effect: NoSchedule'
-        )
-        toleration = toleration.replace('"', '\\"').replace("\n", "\\n")
-        param_cmd = (
-            f'[{{"op": "replace", "path": "/data/CSI_PLUGIN_TOLERATIONS", "value": "{toleration}" }}, '
-            f'{{"op": "replace", "path": "/data/CSI_PROVISIONER_TOLERATIONS", "value": "{toleration}" }}]'
+        # Add tolerations to the subscription
+        sub_list = ocp.get_all_resource_names_of_a_kind(kind=constants.SUBSCRIPTION)
+        param = (
+            '{"spec": {"config":  {"tolerations": '
+            '[{"effect": "NoSchedule", "key": "xyz", "operator": "Equal", '
+            '"value": "true"}]}}}'
         )
-        configmap_obj.patch(params=param_cmd, format_type="json")
+        for sub in sub_list:
+            sub_obj = ocp.OCP(
+                resource_name=sub,
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                kind=constants.SUBSCRIPTION,
+            )
+            sub_obj.patch(params=param, format_type="merge")
 
         # After edit noticed few pod respins as expected
         assert wait_for_pods_to_be_running(timeout=600, sleep=15)
+        self.sanity_helpers.health_check()
+
+        # Check toleration on pods under openshift-storage
+        check_toleration_on_pods(toleration_key="xyz")
 
-        # Respin all pods and check it if is still running
+        # check all pods are present
         pod_list = get_all_pods(
             namespace=defaults.ROOK_CLUSTER_NAMESPACE,
         )
-        for pod in pod_list:
-            pod.delete(wait=False)
-
-        assert wait_for_pods_to_be_running(timeout=600, sleep=15)
-        self.sanity_helpers.health_check()
+        pod_name_list = [
+            pod.get().get("metadata").get("generateName") for pod in pod_list
+        ]
+        logger.info(pod_name_list)
+
+        odf_pods = [
+            "ocs-operator-",
+            "ocs-metrics-exporter-",
+            "odf-operator-controller-manager-",
+            "odf-console-",
```

Comment:
A few of the pods do not exist on the old osd/odf version, for instance, odf-console pod is not present in 4.8. We need to handle those scenarios as well

Reply:
Handling this scenario maybe difficult, for every new/old version few pods added/removed. Everytime the code can break. Instead check for the pod count before and after, That should be enough. Its applicable for all versions

Reply:
Removed this part and added check for the pod count before and after as everytime